python/tskit/trees.py

```diff
@@ -5859,6 +5877,111 @@ def trim(self, record_provenance=True):
         tables.trim(record_provenance)
         return tables.tree_sequence()
 
+    def decapitate(self, time, record_provenance=True):
+        """
+        Return a copy of this tree sequence with topology and mutation information
+        at least as old as the specified time removed. Please see the
+        :meth:`.TableCollection.decapitate` method for details.
+
+        :param float time: The cutoff time.
+        :param bool record_provenance: If True, add details of this operation to the
+            provenance information of the returned tree sequence. (Default: True).
+        :rtype: tskit.TreeSequence
+        """
+        tables = self.dump_tables()
+        tables.decapitate(time, record_provenance=record_provenance)
+        return tables.tree_sequence()
+
+    def split_edges(self, time, *, flags=None, population=None, metadata=None):
```

Comment:
@petrelharp I think I have a workable definition of split_edges here - would you mind taking a look please? The idea would be to push this operation down into C next. The default value stuff is very tedious, but it would have been worse to be stuck with broken semantics later on.

Reply:
Looks good! I've thought carefully and don't see any issues. Note: this has the property that in the result any mutation with time `>= time` will now be above a node whose time is also `>= time`, even in the corner cases.  This convinced me that the strictness of the inequalities was correct.

Reply:
Great, thanks!